Userland/Libraries/LibUnicode/UnicodeUtils.h

```diff
@@ -15,9 +15,16 @@
 
 namespace Unicode::Detail {
 
+enum class TrailingCodePointTransformation {
```

Comment:
Everything looks good, except this does not belong in the Detail namespace - this namespace implies it is an internal implementation detail that outside callers should not use. So it wouldn't make sense for LibWeb to be reaching into Unicode::Detail.

I'm guessing you put this here because the methods in this header need the enum definition? I think what you can do instead is either:

A. Forward-declare this in LibUnicode/Forward.h, then define it in LibUnicode/CharacterTypes.h. You then won't be able to provide a default value for `build_titlecase_string`, but that's fine because it's an internal method and only has one other caller in LibUnicode/String.cpp (which can pass in Lowercase). 

or,

B. Define the enum in the same header (LibUnicode/UnicodeUtils.h), but just outside of the `Detail` namespace (that is, just put a `namespace Unicode { enum class TrailingCodePointTransformation { ... }; }` above the `namespace Unicode::Detail` block).

Reply:
@trflynn89 Thanks for your help! I chose option A because that's what I was going for originally. I hope this is acceptable now.